source/geometry_model/spherical_shell.cc

```diff
@@ -337,6 +354,18 @@ namespace aspect
       return std::map<std::string,types::boundary_id>();
     }
 
+    if (periodic)
```

Comment:
What are you trying to do here? You can not define a function depending on an ``if`` statement.

You probably want this inside the function around the ``.insert`` call.